### PR TITLE
Makes industrial gold and regenerative gold spawn random coins. Also adds chococoin to coin spawner.

### PR DESCRIPTION
**Pull request by Helg2**
## About The Pull Request
So they just had list from 6 coins and not random picking of coins, so now its random.
It will also spawn chococoin because it's a coin too.
But won't spawn `coin/gold/debug` because it's debug and eldritch coin because comxy asked about it.

Added chococoin to `/obj/effect/spawner/random/entertainment/coin` and holdables list of wallet.

## Why It's Good For The Game
Random is funny.

## Changelog

:cl:
add: Indusrial Gold and Regenerative Gold extracts now spawn random coins and not just list of 6 coins. But no mythril as it has miserable chance to have summoning effect which have even more miserable chance to spawn something dangerous and not just mice.
add: Chococoin now has a chance to spawn wherever the coins are used to spawn (wallets, greed ruins, deltastation bar)
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
there's a reason it did that: mythril and adamantine coins. I don't renember if you can melt them back down in materials so adamantine's probably fine, but mythril coins have mythril stats so you could potentially get some insanely powerful currency to throw at people. And we all know fistful of dollar is inferior to ricoshot.

**Comment by Helg2** [comment]
> but mythril coins have mythril stats so you could potentially get some insanely powerful currency to throw at people. And we all know fistful of dollar is inferior to ricoshot.

Can you rephrase so i don't need to spend 5 hours on decoding what you meant?

**Comment by CRITAWAKETS** [comment]
> > but mythril coins have mythril stats so you could potentially get some insanely powerful currency to throw at people. And we all know fistful of dollar is inferior to ricoshot.
> 
> Can you rephrase so i don't need to spend 5 hours on decoding what you meant?

anything made of mythril has rpg loot stats and this includes coins so you get things like coins with +13 force and incindiary effects

**Comment by ATH1909** [comment]
> anything made of mythril has rpg loot stats and this includes coins so you get things like coins with +13 force and incindiary effects

which I think is cool, personally

**Comment by Helg2** [comment]
> anything made of mythril has rpg loot stats and this includes coins so you get things like coins with +13 force and incindiary effects

Considering that it can also be -13 and drops like twice in century is it really that powerfull?

**Comment by the-orange-cow** [comment]
The REAL xeno power gamer rolls for gatfruit with industrial silver.

**Comment by HoodMilk** [comment]
Does it spawn chocolate coins too?

**Comment by Helg2** [comment]
> Does it spawn chocolate coins too?

I'll try to add it.

**Comment by atlasle** [comment]
> > Considering that it can also be -13 and drops like twice in century is it really that powerfull?
> 
> all it takes is 1 guy getting "Mythril coin of summoning Elite Syndicates / Maulers / Leaper Frogs" to ruin everything

okay then it just needs to be hardcoded that like mythril coins only appear 0.01% of the time, no need to throw away the entire feature

**Comment by Helg2** [comment]
> all it takes is 1 guy getting "Mythril coin of summoning Elite Syndicates / Maulers / Leaper Frogs" to ruin everything

Mythril coins also has a chance to spawn in `/obj/effect/spawner/random/entertainment/coin` so eh

**Comment by Comxy** [comment]
The eldritch coin exists which is only meant to be summoned through a heretic. You might want to add that one to the blocked list.

**Comment by Helg2** [comment]
> The eldritch coin exists which is only meant to be summoned through a heretic. You might want to add that one to the blocked list.

Since it's random, this is fine.

**Comment by Elli-Skala** [comment]
Seeing how theses can be fully automated even if rare it can still be brute forced for myhtral coins

**Comment by timothymtorres** [comment]
Not sure why you aren't using the random spawner for coins:

https://github.com/tgstation/tgstation/blob/9f30886fc8ed48dec60106a5bf6b405096883a64/code/game/objects/effects/spawners/random/entertainment.dm#L48-L65

**Comment by Helg2** [comment]
> Not sure why you aren't using the random spawner for coins

It doesn't have all the variations of coins.
And also havs not equal chances.

**Comment by timothymtorres** [comment]
> It doesn't have all the variations of coins. And it also has not equal chances.

Feel free to add the chocolate coin to the spawner.  Yet the more exotic materials are supposed to be rare.  This spawner is also used at roundstart across different locations.

**Comment by Helg2** [comment]
> Seeing how theses can be fully automated even if rare it can still be brute forced for myhtral coins

Considering that rpg loot event could be chosen on april fools and it can be casted by admins, have this ever happened? What are the chances even.

**Comment by MercuryArrow** [comment]
> 

As a Coin Hunting Enjoyer(tm), it's important to remember that many of the RPG loot effects either don't work on coins at all or are busted and only apply their effects in inconsistent ways - I've been disappointed many, many times by 'pyromantic coin of the robustman +6'. The most lucky for the user situation in most cases is that the xenobiologist now has a weapon on par with a welder. Which is to say, they have one about as strong as literally anything that anyone else can grab at roundstart. This is not a meaningful concern.

You also have to factor in that the actual INTENDED way of the crew getting access to mythril, the legendary mining skill rock veins, have been broken for nearly two years following Lavaland mapgen going fucky wucky and that particular intended method yielded actual mythril that could be shoved in a lathe and used to make other things that *do* inherit the RPG stats properly.

Put simply, the absolute worst case scenario for mythril coins is that a xenobiologist who has expended their time and effort on getting their hands on a mythril coin now has something that is in every way equal to what most other crew had to begin with, and actually worse than the previously-established method of the crew getting their hands on the material.  It's fine.

**Comment by KittyNoodle** [comment]
> > > but mythril coins have mythril stats so you could potentially get some insanely powerful currency to throw at people. And we all know fistful of dollar is inferior to ricoshot.
> > 
> > 
> > Can you rephrase so i don't need to spend 5 hours on decoding what you meant?
> 
> anything made of mythril has rpg loot stats and this includes coins so you get things like coins with +13 force and incindiary effects

Mythril coins can also be acquired in metastation bar rarely, so blacklisting them here would remove that too. A thing I would like to stay on account of being a funny occurrence when the coin ends up summoning a statue or some shit directly in the bar

**Comment by Mothblocks** [comment]
> > The eldritch coin exists which is only meant to be summoned through a heretic. You might want to add that one to the blocked list.
> 
> Since it's random, this is fine.

What lol?

**Comment by Helg2** [comment]
> What lol?

I meant like that it's pretty useless for non-heretics and not even guaranteed to spawn. And besides that it's usefull only for heretics so does it even matters?

**Comment by Helg2** [comment]
@Comxy Since we're here is it meant to be `put` or `insert`?
https://github.com/tgstation/tgstation/blob/ba4e472e89cf546812718cda1ec2d9ff93fdb490/code/modules/mining/ores_coins.dm#L593-L597

**Comment by Helg2** [comment]
> > What lol?
> 
> I meant like that it's pretty useless for non-heretics and not even guaranteed to spawn. And besides that it's usefull only for heretics so does it even matters?

The only way i can see it being used properly is to antag check heretics but that's on the same level as checking cultists via attacking someone with dagger.

**Comment by Comxy** [comment]
Insert was the intended wording.

Also my point was that being able to obtain the coin through other means than heretic would undermine the way to obtain the item. Even though the coin isn't that good and more of a joke, until it gets replaced with a different research node I don't think it should ever be spawnable outside of heretics.

**Comment by Helg2** [comment]
eh, okay

File: code/modules/mining/ores_coins.dm [hunk 1]
```diff
@@ -592,7 +592,7 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
 		return
 	if(istype(target_atom, /obj/machinery/door/airlock))
 		var/obj/machinery/door/airlock/target_airlock = target_atom
-		to_chat(user, span_warning("You put insert the [src] into the airlock."))
+		to_chat(user, span_warning("You insert the [src] into the airlock."))
```
Comment (Fikou) [inline code/modules/mining/ores_coins.dm:595]: if youre changing this already
```suggestion
		to_chat(user, span_warning("You insert [src] into the airlock."))
```

**Review by Fikou — CHANGES_REQUESTED** [review]
mithril coin is not happening

**Comment by Helg2** [comment]
> mithril coin is not happening

Why

**Comment by Mothblocks** [comment]
fikou doesnt want to make introducing wizard rpg loot into normal rounds a more regular thing than it is now. i think that is reasonable.

image

**Comment by Helg2** [comment]
> fikou doesnt want to make introducing wizard rpg loot into normal rounds a more regular thing than it is now. i think that is reasonable.
> 
> image

Has it ever happened or he's just being conservative?

**Comment by Mothblocks** [comment]
do you plan on doing it or not

**Comment by Mothblocks** [comment]
No

**Comment by Helg2** [comment]
Why

File: code/modules/research/xenobiology/crossbreeding/industrial.dm [hunk 2]
```diff
@@ -166,7 +166,7 @@ Industrial extracts:
 
 /obj/item/slimecross/industrial/gold
 	colour = "gold"
-	effect_desc = "Produces random coins."
+	effect_desc = "Produces random coins. But no mythril, as it's to magical for something magical already."
```
Comment (Mothblocks) [inline code/modules/research/xenobiology/crossbreeding/industrial.dm:169]: We probably don't need to clarify this--we don't give eldritch coins either.

Comment (Helg2) [reply code/modules/research/xenobiology/crossbreeding/industrial.dm:169]: But the reasons are different
image

**Comment by Loafin34** [comment]
I can't really figure this out in the code, but is the Antag Token and the TGC Flipper in here? (antag token from arcade machines that lands on valid or salad, and tgc flipper from the tgc holodeck game.)

**Comment by Helg2** [comment]
> I can't really figure this out in the code, but is the Antag Token and the TGC Flipper in here? (antag token from arcade machines that lands on valid or salad, and tgc flipper from the tgc holodeck game.)

Yes
